README.md

```diff
@@ -91,6 +96,21 @@ initIpld('/tmp/ifpsrepo', (err, ipld) => {
 
 ## API
 
+The IPLD API works strictly with CIDs and deserialized IPLD Nodes. Interacting with the binary data happens on lower levels. To access the binary data directly, use the [Block API](https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/BLOCK.md).
+
+All methods that return an async iterator return one that got extended with convenience methods:
+
+  - `iter.first()`: Return the first item only
+  - `iter.last()`: Return the last item only
+  - `iter.all()`: Return all items as array
```

Comment:
I think that it this is confusing way to communicate. I would suggest to just define a type / class e.g `Batch` and say that it implements [async iteration interface](https://github.com/tc39/proposal-async-iteration) along with methods `first`, `last`, `all`.

That would make communication clear:

1. It's clear that `putMany` takes async iterable, while it does return `Batch`, otherwise it leaves reader wondering if whether `putMany` should also be provided async iterable that has those extensions.
2. `Batch` could be exposed and could have `Batch.from(asyncIterable)` in case users need to create own API compatible instances without needing to do it on their own.
3. In the future `Batch` API can be further extended with other methods.

Reply:
I've opened https://github.com/ipld/js-ipld/issues/196 as kind of a "catch all" issue for batch API related things.